src/status_im/ethereum/transactions/core.cljs

```diff
@@ -119,36 +119,26 @@
                 watched-transactions))))
 
 (fx/defn add-transfer
-  [{:keys [db] :as cofx} {:keys [hash id] :as transfer}]
-  (let [transfer-by-hash (get-in db [:wallet :transactions hash])
-        transfer-by-id   (get-in db [:wallet :transaction id])
-        unique-id (when-not (or transfer-by-id
-                                (= transfer transfer-by-hash))
-                    (if (and transfer-by-hash
-                             (not (= :pending
-                                     (:type transfer-by-hash))))
-                      id
-                      hash))]
-    (when unique-id
+  [{:keys [db] :as cofx} {:keys [hash id] :as transfer} address]
+  (let [transfer-by-hash (get-in db [:wallet :accounts address :transactions hash])]
+        ;;transfer-by-id   (get-in db [:wallet :transaction id]) ;; TODO didn't found any usage of this
```

Comment:
some transactions contain more than one transfer and are deduped by id since they have the same hash. that is definitely used